_develop/connect/use-with-clarity.md

```diff
@@ -0,0 +1,92 @@
+---
+layout: learn
+permalink: /:collection/:path.html
+---
+# Usage with Clarity
+{:.no_toc}
+
+With Connect, you can interact with the Stacks 2.0 blockchain. You can allow your users to send transactions and interact with smart contracts.
+
+* TOC
+{:toc}
+
+## How it works
+
+For your app's users to be able to execute a smart contract function, they need to sign and broadcast a transaction. It's important that users remain control of the private keys that sign these transactions. Connect provides an easy-to-use workflow that allows your users to securely sign transactions.
+
+Connect allows you to open the authenticator with parameters indicating the details of the transaction - like the smart contract address, function name, and specific arguments. Your users get the chance to see these details, and then sign and broadcast the transaction in a single click. Their transaction will be securely signed and broadcasted onto the Stacks blockchain. After this is done, a callback is fired to allow you to update your app.
+
+## API
+
+### `ContractCallOptions`
+
+When signing a transaction, you need to specify a few details. Here is the exact interface that describes what options you have:
+
+```ts
+export interface ContractCallOptions {
+  contractAddress: string;
+  functionName: string;
+  contractName: string;
+  functionArgs?: any[];
+  authOrigin?: string;
+  userSession?: UserSession;
+  appDetails?: AuthOptions['appDetails'];
+  finished?: (data: FinishedTxData) => void;
+}
+```
+
+- `contractAddress`: The Stacks address that published this contract
+- `contractName`: The name that was used when publishing this contract
+- `functionName`: The name of the function you're calling. This needs to be a [public function](/core/smart/clarityRef.html#define-public).
+- `functionArgs`: The arguments you're calling the function with
+- `authOrigin`: _(optional)_ The URL for the authenticator. You only need to use this if you're developing the authenticator locally. Do not use in production
+- `userSession`: _(optional)_ A `UserSession` instance
+- `appDetails`: A dictionary that includes `name` and `icon`
+- `finished`: A callback that is fired when the transaction is signed and broadcasted. Your callback will receive an object back with a `txId` and a `txRaw`.
+
+### Usage in ES6 (non-React) apps
+
+```ts
+import { openContractCall } from '@blockstack/connect';
+
+// Here's an example of options:
+const options = {
+  contractAddress: 'ST22T6ZS7HVWEMZHHFK77H4GTNDTWNPQAX8WZAKHJ',
```

Comment:
Just a thought: should Blockstack create a handful of test/example/demo stacks addresses for which the org holds the keys. 

This'd be one layer of protection again some silly mistakes.

Reply:
Yeah, not a terrible idea!

Reply:
The pro move would be to auto-siphon any mainnet funds from those addresses into a dev fund :p

Reply:
Hank's "burnt" crypto address